<launch>
	<!--env name="ROSCONSOLE_CONFIG_FILE" value="$(find next_best_view)/log/rosconsole.config" /-->
	<include file="$(find asr_mild_navigation)/launch/simulation/simulation_floor.launch" />
	<include file="$(find world_model)/launch/world_model.launch" />
	<include file="$(find recognition_for_grasping)/launch/recognition_manager.launch" />
	<include file="$(find asr_flir_ptu_driver)/launch/ptu_left_mock.launch" />
	<include file="$(find next_best_view)/launch/RobotModelServiceSim.launch" />
	<node pkg="next_best_view" type="next_best_view_parameters_test" name="nbv" output="screen">

		<param name="output_path" value="$(find next_best_view)/rsc/" />
	
		<rosparam file="$(find next_best_view)/rsc/next_best_view_settings_sim.yaml"/>
          	<rosparam file="$(find next_best_view)/rsc/VisualizationConfig.yaml"/>
		<rosparam file="$(find next_best_view)/rsc/TopicsConfig.yaml"/>
                <rosparam file="$(find next_best_view)/rsc/costmap_params.yaml" command="load" ns="global_costmap"/>
                <rosparam file="$(find asr_mild_navigation)/rsc/global_costmap_params.yaml" command="load"/>
                <param name="mCropBoxListFilePath" value="$(find next_best_view)/rsc/CropBoxListHighResolution.xml"/>
                <param name="mMapsImagePath" value="$(find next_best_view)/rsc"/>

		<param name="side_objects/test" value="true"/>
		<rosparam param="side_objects/initial_pose">[24.268 , 9.612 , 1.32 , 0.0 , 0.0 , 0.478 , 0.878]</rosparam>
		<rosparam param="side_objects/element_orientation">[-90.0 , 90.0 , 0.0]</rosparam>
		<rosparam param="side_objects/element_deviation">[0.0, 0.0, 0.0]</rosparam>
		<param name="side_objects/element_type" value="CeylonTea"/>
		<param name="side_objects/cloud_size" value="1"/>
		<param name="side_objects/step_size" value="5"/>
		<param name="side_objects/step_number" value="10"/>
		
		<param name="configurable_point_clouds/test" value="false"/>
		<param name="positions/test" value="false"/>
		
		<rosparam file="$(find asr_robot_model_services)/rsc/robot_model_services_settings_sim.yaml"/>
		<rosparam file="$(find asr_robot_model_services)/rsc/robot_model_frames.yaml"/>
	  	<rosparam file="$(find asr_robot_model_services)/rsc/TopicsConfig.yaml"/>
	</node>
</launch>
